samcli/commands/check/lambda_function_pricing_calculations.py

```diff
@@ -0,0 +1,227 @@
+"""
+
+from typing import List
+import urllib.request
+from urllib.error import HTTPError
+
+import ast
+
+from botocore.session import get_session
+
+from samcli.commands.check.pricing_calculations import PricingCalculations
+
+
+class LambdaFunctionPricingCalculations(PricingCalculations):
+    def __init__(self, graph):
+        super().__init__(graph)
+        self._max_num_of_free_requests = None
+        self._max_free_gb_s = None
+        self._monthly_compute_charge = None
+        self._monthly_request_charge = None
+        self._max_request_usage_type = "Global-Request"
+        self._max_free_gb_s_usage_type = "Global-Lambda-GB-Second"
+        self._compute_usage_type = "Lambda-GB-Second"
+        self._request_charge_usage_type = "Request"
+
+        self.lambda_pricing_results = None
+
+    def get_aws_pricing_info(self, region):
+        return _get_aws_lambda_pricing_info(region)
+
+    def determine_cost(self) -> None:
+        self._determine_lambda_cost()
+
+    def run_calculations(self) -> None:
+        self.get_charge_and_request_amounts()
+        self.determine_cost()
+
+    def get_charge_and_request_amounts(self) -> None:
+        self._get_lambda_charge_and_request_amounts()
+
+    def _determine_lambda_cost(self) -> None:
+        """
+        Determines the cost of running all lambda functions based on the
+        answers the user provided for lamnda function pricing
+        """
+        template_lambda_pricing_info = self.graph.unique_pricing_info["LambdaFunction"]
+
+        memory_amount = float(template_lambda_pricing_info.allocated_memory)
+        memory_unit = template_lambda_pricing_info.allocated_memory_unit
+        number_of_requests = template_lambda_pricing_info.number_of_requests
+        average_duration = template_lambda_pricing_info.average_duration
+
+        # Need to convert to GB if provided in MB
+        if memory_unit == "MB":
+            memory_amount *= 0.0009765625
+
+        # converts average_duration from ms to seconds within equation
+        total_compute_s = number_of_requests * average_duration * 0.001
+
+        total_compute_gb_s = memory_amount * total_compute_s
+
+        total_compute_gb_s -= self._max_free_gb_s
+
+        if total_compute_gb_s < 0:
+            total_compute_gb_s = 0
+
+        monthly_compute_amount = total_compute_gb_s * self._monthly_compute_charge
+
+        total_requests = number_of_requests - self._max_num_of_free_requests
+
+        if total_requests < 0:
+            total_requests = 0
+
+        monthly_request_amount = total_requests * self._monthly_request_charge
+
+        monthly_lambda_costs = round(monthly_compute_amount + monthly_request_amount, 2)
+
+        self.lambda_pricing_results = monthly_lambda_costs
+
+    def _get_lambda_charge_and_request_amounts(self) -> None:
+        """
+        Gets the charge and request amouint from the AWS Price List API
+        """
+        region = get_session().get_config_variable("region")
```

Comment:
Is there any chance that it will be None? (e.g. when no default region is set in the user's aws cli profile?)